core/model/src/main/java/org/eclipse/rdf4j/model/datatypes/XMLDatatypeUtil.java

```diff
@@ -676,6 +681,15 @@ public static boolean isValidAnyURI(String value) {
 		}
 	}
 
+	public static boolean isValidLanguage(String value) {
```

Comment:
Currently in core/rio/languages:

https://github.com/eclipse/rdf4j/blob/master/core/rio/languages/src/main/java/org/eclipse/rdf4j/rio/languages/BCP47LanguageHandler.java#L38

Advantage of the existing version was that it distinguished BCP47 from the previous RFC3066, but everything should have switched over to RDF-1.1 by now so the abstraction there may be a bit redundant now.

XMLDatatypeUtil isn't the ideal location for long term reuse of this particular function, but would be good to have it in core/model somewhere.